### PR TITLE
Add ability to save policies with BC

### PR DESCRIPTION
**Pull request by qxcv**
This allows the behavioural cloning class to save and reload policies. It'd be nice if there were some unified way of saving policies, but I think that would require upstream changes to stable baselines (e.g. to _always_ save the kwargs policies were created with so that they can be reconstructed).

### CODE REVIEW

**Comment by AdamGleave** [comment]
Stable Baselines places the onus of loading/saving in the RL algorithm, i.e. they have to override `load()` and `save()` in `BaseRLModel`. It might be nice to follow that kind of interface, we could use `BaseRLModel._save_to_file` and `BaseRLModel._load_from_file`, but it's not *that* helpful.

**Comment by AdamGleave** [comment]
Also you need to merge a recent version of `origin/master` into this branch (it's old enough it doesn't have the CircleCI config).

File: tests/test_bc.py [hunk 1]
```diff
@@ -1,22 +1,50 @@
...
+  # just to change the values a little
+  bc_trainer.train(n_epochs=1)
+  var_values = bc_trainer.sess.run(bc_trainer.policy_variables)
+  bc_trainer.save_policy(pol_path)
+  with tf.Session() as sess:
+    # just make sure it doesn't die
+    with tf.variable_scope('new'):
+      bc.BCTrainer.reconstruct_policy(pol_path)
+    new_vars = tf.get_collection(
+        tf.GraphKeys.TRAINABLE_VARIABLES, scope='new')
```
Comment (shwang) [inline tests/test_bc.py]: Probably cleaner to use use `reconstructed_policy.policy_variables`. (Not sure if this is possible.)

File: setup.py [hunk 2]
```diff
@@ -61,6 +62,7 @@
             'isort',
             'jupyter',
             'pytype',
+            'codespell',
```
Comment (shwang) [inline setup.py]: Merge note: I moved `codespell`, `pytype` to TESTS_REQUIRE on master.

**Comment by AdamGleave** [comment]
@shwang @qxcv are we blocked on anything here or should someone merge this?

File: tests/test_bc.py (hunk 1 above)
Comment (shwang) [reply tests/test_bc.py]: (Was mistaken here, as `reconstruct_policy` returns a policy not a BCTrainer)

**Comment by shwang** [comment]
Forgot about this, sorry. Will merge after tests pass

**Pull request merged**